### DISCUSSION TITLE
Can OME act as a pure signalling server?

### DISCUSSION THREAD

**Discussion by UkuKert** [OvenMediaLabs/OvenMediaEngine#1688] [Help]
Hello! Thank you for this amazing multi-media tool, I love it!

While playing around and learning about WebRTC streams I learned that WebRTC is designed to be peer-to-peer, skipping any media servers. OME has a functionality of publishing WebRTC and receiving WebRTC. It also has a built-in signalling server to negotiate the WebRTC connections. At first glance it seems that it has all the components to facilitate a peer-to-peer WebRTC connection.

However during testing I found out that the WebRTC stream is actually piped through OME, with it acting as a republisher of the stream. If I understand correctly, then instead of having one WebRTC connection between peers, OME establishes 2: Peer -> OME -> Peer.
I also noticed the P2P experiment tab in the docs, however that seemed to be just for forwarding a received stream from 1 OvenPlayer peer to another.

Is my understanding correct?
Is there any other way to do peer-to-peer WebRTC streaming between 2 OvenPlayers for example?
If no, would it make sense to implement this functionality in OME/OvenPlayer so it would just act as a signalling server and we could have true peer-to-peer connections? It seems to me that all the components for this exist, it is just the matter of assembling them correctly.

Best regards,
Uku

**Comment by bchah** [comment]
I don't know the complete answer, but I know that the experimental P2P functionality stopped working last year and there is not currently any plan to revive it: https://github.com/AirenSoft/OvenMediaEngine/issues/1466

My guess is that AirenSoft would suggest use of a pure P2P connectivity tool like [PeerJS](https://peerjs.com) instead of OME, given the proprietary nature of their WebRTC implementation. You may also require a TURN server like [coturn](https://github.com/coturn/coturn).

**Comment by getroot** [comment]
OME acts as an SFU or MCU from the perspective of a WebRTC terminal. Play around with WebRTC SFUs or MCUs. Then you will understand OME. OME even supports protocols such as RTMP/SRT in addition to WebRTC input/output, and also supports a built-in transcoder.
